packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts

```diff
@@ -49,6 +49,205 @@ test('has many query', () => {
+});
+
+test('fails if property does not exist on related type with references', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: [Member] @hasMany(references: ["teamID"])
+    }
+    type Member @model {
+      id: ID!
+      team: Team @belongsTo(references: ["teamID"])
+    }`;
+
+  expect(() =>
+    testTransform({
+      schema: inputSchema,
+      transformers: [new ModelTransformer(), new IndexTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError('teamID is not a field in Member');
+});
+
+test('fails if an empty list of references is passed in', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: [Member] @hasMany(references: [])
+    }
+    type Member @model {
+      id: ID!
+      team: Team @belongsTo(references: [])
+    }`;
+
+  expect(() =>
+    testTransform({
+      schema: inputSchema,
+      transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError('Invalid @hasMany directive on members - empty references list');
+});
+
+test('fails if related type does not exist', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: [Foo] @hasMany(references: ["teamID"])
+    }
+    type Member @model {
+      id: ID!
+      teamID: String
+      team: Team @belongsTo(references: ["teamID"])
+    }`;
+
+  expect(() =>
+    testTransform({
+      schema: inputSchema,
+      transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError(); // TODO: Fix validation error messaging - 'Unknown type "Foo". Did you mean "Member"?'
+});
+
+test('fails if hasMany related type is not an array', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: Member @hasMany(references: ["teamID"])
+    }
+    type Member @model {
+      id: ID!
+      teamID: String
+      team: Team @belongsTo(references: ["teamID"])
+    }`;
+
+  expect(() =>
+    testTransform({
+      schema: inputSchema,
+      transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError(); // TODO: Error message
+});
+
+test('fails if uni-directional hasMany', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: [Member] @hasMany(references: ["teamID"])
+    }
+    type Member @model {
+      id: ID!
+      teamID: String
+      team: Team
+    }`;
+
+  // TODO: This should fail -- find appropriate place to validate.
+  // expect(() =>
+  //   testTransform({
+  //     schema: inputSchema,
+  //     transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+  //   }),
+  // ).toThrowError();
+});
```

Comment:
nit: are there future work tickets to implement these validations?

Reply:
Just these TODOs and the [feature branch PRs description](https://github.com/aws-amplify/amplify-category-api/pull/2370) for larger things. There are quite a few of them in the feature branch already.